packages/node-runtime-worker-thread/src/worker-runtime.spec.ts

```diff
@@ -40,23 +62,329 @@ describe('worker', () => {
+              // @ts-ignore
+              .bson(new ObjectId('000000000000000000000000'));
+            expect(printable).to.have.nested.property('[0].foo', 321);
+          }
+        ],
+        [
+          [
+            `use ${db}`,
+            'db.coll.insertMany([1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map(i => ({ i })))',
+            'db.coll.find({ i: { $mod: [2, 0] } }, { _id: 0 })'
+          ],
+          'Cursor',
+          [{ i: 2 }, { i: 4 }, { i: 6 }, { i: 8 }, { i: 10 }]
+        ],
+        [
+          [
+            `use ${db}`,
+            "db.coll.insertMany('a'.repeat(100).split('').map(a => ({ a })))",
+            'db.coll.find({}, { _id: 0 })',
+            'it'
+          ],
+          'CursorIterationResult',
+          ({ printable }: ShellResult) => {
+            expect(printable).to.include.deep.members([{ a: 'a' }]);
+          }
+        ]
+      ];
+
+      const crudCommands: CommandTestRecord[] = [
+        [
+          [`use ${db}`, 'db.coll.insertOne({ a: "a" })'],
+          'InsertOneResult',
+          ({ printable }: ShellResult) => {
+            expect(printable).to.have.property('acknowledged', true);
+            expect(printable)
+              .to.have.property('insertedId')
+              .instanceof(ObjectId);
+          }
+        ],
+        [
+          [`use ${db}`, 'db.coll.insertMany([{ b: "b" }, { c: "c" }])'],
+          'InsertManyResult',
+          ({ printable }: ShellResult) => {
+            expect(printable).to.have.property('acknowledged', true);
+            expect(printable)
+              .to.have.nested.property('insertedIds[0]')
+              .instanceof(ObjectId);
+          }
+        ],
+        [
+          [
+            `use ${db}`,
+            'db.coll.insertOne({ a: "a" })',
+            'db.coll.updateOne({ a: "a" }, { $set: { a: "b" } })'
+          ],
+          'UpdateResult',
+          {
+            acknowledged: true,
+            insertedId: null,
+            matchedCount: 1,
+            modifiedCount: 1,
+            upsertedCount: 0
+          }
+        ],
+        [
+          [
+            `use ${db}`,
+            'db.coll.insertOne({ a: "a" })',
+            'db.coll.deleteOne({ a: "a" })'
+          ],
+          'DeleteResult',
+          { acknowledged: true, deletedCount: 1 }
+        ],
+        [
+          [`use ${db}`, 'db.coll.bulkWrite([{ insertOne: { d: "d" } }])'],
+          'BulkWriteResult',
+          ({ printable }: ShellResult) => {
+            expect(printable).to.have.property('acknowledged', true);
+            expect(printable).to.have.property('insertedCount', 1);
+            expect(printable)
+              .to.have.nested.property('insertedIds[0]')
+              .instanceof(ObjectId);
+          }
+        ]
+      ];
+
+      showCommand
+        .concat(useCommand)
+        .concat(helpCommand)
+        .concat(cursors)
+        .concat(crudCommands)
```

Comment:
I tried to come up with as many special `result`s to test here, any other types we should cover with tests?

Reply:
I think those should be fine, My biggest hope was that all of them where to be serialized in the exact same way without exceptions in the shell api, and i still think that's what we should aim for, but is ok if we try to improve it in a second moment.